awx_collection/tests/integration/targets/tower_lookup_api_plugin/tasks/main.yml

```diff
@@ -0,0 +1,243 @@
+
+    - assert:
+        that:
+          - user_objects | length() == 1
+          - users_list['results'][0]['id'] == user_objects[0]['id']
+
+    - name: Loop over one user with the loop syntax
+      assert:
+        that:
+          - item['id'] == user_creation_results['results'][0]['id']
+      loop: "{{ query(plugin_name, 'users', query_params={ 'username' : user_creation_results['results'][0]['item'] } ) }}"
+      loop_control:
+        label: "{{ item.id }}"
+
+    - name: Get a page of users as just ids
+      set_fact:
+        users: "{{ query(plugin_name, 'users', query_params={ 'username__endswith': test_id, 'page_size': 2 }, return_ids=True ) }}"
+
+    - name: Assert that user list has 2 ids only and that they are strings, not ints
+      assert:
+        that:
+          - users | length() == 2
+          - user_creation_results['results'][0]['id'] not in users
+          - user_creation_results['results'][0]['id'] | string in users
+
+    - name: Get all users of a system through next attribute
+      set_fact:
+        users: "{{ query(plugin_name, 'users', query_params={ 'username__endswith': test_id, 'page_size': 1 }, return_all=true ) }}"
+
+    - assert:
+        that:
+          - users | length() >= 3
+
+    - name: Get all of the users created with a max_objects of 1
+      set_fact:
+        users: "{{ lookup(plugin_name, 'users', query_params={ 'username__endswith': test_id, 'page_size': 1 }, return_all=true, max_objects=1 ) }}"
+      ignore_errors: true
+      register: max_user_errors
+
+    - assert:
+        that:
+          - max_user_errors is failed
+          - "'List view at users returned 3 objects, which is more than the maximum allowed by max_objects' in max_user_errors.msg"
+
+    - name: Get the ID of the first user created and verify that it is correct
+      assert:
+        that: "{{ query(plugin_name, 'users', query_params={ 'username' : user_creation_results['results'][0]['item'] }, return_ids=True)[0] }} == {{ user_creation_results['results'][0]['id'] }}"
+
+    - name: Try to get an ID of someone who does not exist
+      set_fact:
+        failed_user_id: "{{ query(plugin_name, 'users', query_params={ 'username': 'john jacob jingleheimer schmidt' }, expect_one=True) }}"
+      register: result
+      ignore_errors: true
+
+    - assert:
+        that:
+          - result is failed
+          - "'Expected one object from endpoint users' in result['msg']"
+
+    - name: Lookup too many users
+      set_fact:
+        too_many_user_ids: " {{ query(plugin_name, 'users', query_params={ 'username__endswith': test_id }, expect_one=True) }}"
+      register: results
+      ignore_errors: true
+
+    - assert:
+        that:
+          - results is failed
+          - "'Expected one object from endpoint users, but obtained 3' in results['msg']"
+
+    - name: Get the settings page
+      assert:
+        that:
+          - "'CUSTOM_LOGO' in {{ lookup(plugin_name, 'settings/ui' ) }}"
```

Comment:
It looks like this caused some new kind of test failure

```
  TASK [tower_lookup_api_plugin : Get the settings page] *************************
  fatal: [testhost]: FAILED! => {"msg": "The conditional check ''CUSTOM_LOGO' in {u'PENDO_TRACKING_STATE': u'off', u'MAX_UI_JOB_EVENTS': 4000, u'CUSTOM_LOGO': u'', u'UI_LIVE_UPDATES_ENABLED': True, u'CUSTOM_LOGIN_INFO': u''}' failed. The error was: template error while templating string: expected token ':', got 'string'. String: {% if 'CUSTOM_LOGO' in {u'PENDO_TRACKING_STATE': u'off', u'MAX_UI_JOB_EVENTS': 4000, u'CUSTOM_LOGO': u'', u'UI_LIVE_UPDATES_ENABLED': True, u'CUSTOM_LOGIN_INFO': u''} %} True {% else %} False {% endif %}"}
```

I don't quite know what that means, but this does seem duplicated with the other task that saves the result to `tower_settings`.

Also looks like this might be related to a bug in the assert module...

Reply:
It is a duplicate but it's odd that its failing in your tests (it passes in mine locally). I wonder if its a python/ansible version issue? Either way, we can remove this one.